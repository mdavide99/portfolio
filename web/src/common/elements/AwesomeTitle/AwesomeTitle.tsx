import classNames from "classnames";

interface AwesomeTitleProps {
  content: String;
  size?: "sm" | "md" | "lg";
  className?: string;
}

function AwesomeTitle(props: AwesomeTitleProps) {
  let size: "sm" | "md" | "lg";
  if (typeof props.size === "undefined") {
    size = "lg";
  } else {
    size = props.size;
  }

  let titleClass = classNames(
    props.className,
    "font-extrabold  text-transparent bg-clip-text bg-gradient-to-t from-blue-600 to-blue-400  dark:bg-gradient-to-t dark:from-fuchsia-600 dark:to-fuchsia-400",
    "text-4xl sm:leading-tight sm:text-5xl sm:leading-tight mb-2 sm:mb-3"
  );

  switch (size) {
    case "sm":
      return <h3 className={titleClass}>{props.content}</h3>;

    case "md":
      return <h2 className={titleClass}>{props.content}</h2>;

    default:
      return <h1 className={titleClass}>{props.content}</h1>;
  }
}

export { AwesomeTitle };
