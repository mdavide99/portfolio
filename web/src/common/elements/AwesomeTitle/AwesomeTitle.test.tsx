import { AwesomeTitle } from "./AwesomeTitle";
import { render, screen } from "@testing-library/react";

describe("AwesomeTitle", () => {
  it("renders AwesomeTitle correctly", () => {
    const textToFind = "Hello World!";
    const { container } = render(<AwesomeTitle content={textToFind} />);
    const renderedText = screen.getByText(textToFind);
    expect(container).toMatchSnapshot();
    expect(renderedText).toBeInTheDocument();
  });
});
