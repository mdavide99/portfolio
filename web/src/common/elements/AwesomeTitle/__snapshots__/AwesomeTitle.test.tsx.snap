// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwesomeTitle renders AwesomeTitle correctly 1`] = `
<div>
  <h1
    class="font-extrabold  text-transparent bg-clip-text bg-gradient-to-t from-blue-600 to-blue-400  dark:bg-gradient-to-t dark:from-fuchsia-600 dark:to-fuchsia-400 text-4xl sm:leading-tight sm:text-5xl sm:leading-tight mb-2 sm:mb-3"
  >
    Hello World!
  </h1>
</div>
`;
